includes/Products/Sync/Background.php

```diff
@@ -282,21 +282,26 @@ private function prepare_product_variation_data( $product ) {
 	 *
 	 * @since 2.0.0
 	 *
-	 * @param array $data product data
+	 * @param array $data product data.
 	 * @return array
 	 */
 	private function normalize_product_data( $data ) {
 
-		// allowed values are 'refurbished', 'used', and 'new', but the plugin has always used the latter
+		// Allowed values are 'refurbished', 'used', and 'new', but the plugin has always used the latter.
 		$data['condition'] = 'new';
 
-		// attributes other than size, color, pattern, or gender need to be included in the additional_variant_attributes field
+		// Attributes other than size, color, pattern, or gender need to be included in the additional_variant_attributes field.
 		if ( isset( $data['custom_data'] ) && is_array( $data['custom_data'] ) ) {
 
 			$attributes = array();
 
 			foreach ( $data['custom_data'] as $key => $val ) {
-				$attributes[] = $key . ':' . $val;
+				$attribute_value = str_replace(
+					',',
+					apply_filters( 'facebook_for_woocommerce_variant_attribute_comma_replacement', ' ' ),
```

Comment:
Suggest adding a docblock for all new hooks or public APIs, explaining the purpose, intended use, parameters & return value, similar to how [WP core documents hooks](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/). This also is a good place to link to the relevant Facebook API/platform docs.

Something like:

```php
/**
 * Filter: facebook_for_woocommerce_variant_attribute_comma_replacement
 *
 * The Facebook API expects a comma-separated list of attributes in `additional_variant_attribute` field.
 * https://developers.facebook.com/docs/marketing-api/catalog/reference/
 * This means that WooCommerce product attributes included in this field should avoid the comma (`,`) character.
 * Facebook for WooCommerce replaces any `,` with a space by default. 
 * This filter allows a site to provide a different replacement string.
 *
 * @since 2.5.0
 *
 * @param string $replacement The default replacement string (`,`).
 * @return string Return the desired replacement string.
 */
```

Reply:
I think also worth passing `$val`, so merchants can choose a different replacement based on it.
```suggestion
					apply_filters( 'facebook_for_woocommerce_variant_attribute_comma_replacement', ' ', $val ),
```